src/otaclient_common/downloader.py

```diff
@@ -343,7 +343,6 @@ def close(self) -> None:
         """
         self._session.close()
 
-    @retry_on_digest_mismatch
```

Comment:
In my understanding, `retry_on_digest_mismatch` is to detect hash mismatch(= download was succeeded, but hash was mismatched). `downloading_watchdog` is for timeout by watch dog.
In this change, my question is that what will be happened when download is succeeded but hash mismatch is occurred? Expected behavior will be to retry.

Reply:
`retry_on_digest_mismatch` implements the `OTA_CACHE_CONTROL` protocol between otaclient and otaproxy.

During downloading, otaproxy will cache the requests from otaclient. If otaclient detects an incomplete/broken downloading by digest check,  `retry_on_digest_mismatch` will inject a header into the request of next downloading request, telling otaproxy to clear the previously cached file.